src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  FormEvent, SetStateAction,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { postComment } from '../utils/fetchClient';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post;
+  postComments: Comment[];
+  setPostComments: Dispatch<SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  selectedPost,
+  setPostComments,
+  postComments,
+}) => {
+  const [fieldName, setFieldName] = useState('');
+  const [fieldEmail, setFieldEmail] = useState('');
+  const [fieldComment, setFieldComment] = useState('');
+  const [validName, setValidName] = useState(false);
+  const [validEmail, setValidEmail] = useState(false);
+  const [validComment, setValidComment] = useState(false);
+  const [isButtonLoad, setIsButtonLoad] = useState(false);
+
+  const nameChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFieldName(event.target.value);
+    setValidName(false);
+  };
+
+  const emailChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFieldEmail(event.target.value);
+    setValidEmail(false);
+  };
+
+  const commentChangeHandler = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setFieldComment(e.target.value);
+    setValidComment(false);
+  };
+
+  const clearHandler = () => {
+    setValidComment(false);
+    setValidName(false);
+    setValidEmail(false);
+    setFieldName('');
+    setFieldEmail('');
+    setFieldComment('');
+  };
+
+  const submitFormHandler = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    let formError = false;
+
+    const newComment = {
+      postId: selectedPost.id,
+      name: fieldName,
+      email: fieldEmail,
+      body: fieldComment,
+    };
+
+    if (fieldName.trim().length === 0) {
```

Comment:
Use logical operators 
```suggestion
    if (!fieldName.trim().length) {
```

Reply:
Done